help/Readme.md

```diff
@@ -0,0 +1,140 @@
+### General
+Virtual-environment project uses packer tool and contains a number of templates for several OS types: Windows 2016/2019, Ubuntu 16.04/18.04/20.04.
+Each image template contains set of scripts that will be executed on virtual machine.
+Packer process will initialize connection to Azure subscription via Azure CLI, and deploy required resources, such as temporary resource group, network interfaces and VM from the "clean" image that specified in template.
+If deployment succeeded, build agent will connect to the deployed VM and begin to execute installation steps from selected the template.
+Connection can be established via SSH or WinRM, depending on image type, so the ports between build agent and temporary VM have to be opened for connection.
+If any template step fails, image generation will be aborted and temporary VM will be terminated.
+If all steps are succeeded, a temporary VM snapshot will be converted to VHD and then uploaded to the specified Azure Storage Account. After that image can be deployed further from this account.
+
+### Build Agent requirements
+- `OS` - Windows/Linux
+- `packer` - Can be downloaded from https://www.packer.io/downloads
+- `PowerShell 5.0 or higher` or `PSCore` for linux distributes.
+- `Azure CLI ` - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
+
+#### Azure DevOps self-hosted pool requirements
+To connect to temporary VM packer will use WinRM, or SSH connections on public IP interfaces.
+If you use build agent located in some Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case if firewall restrictions forbid connections from public addresses, you may deploy private virtual network resources, and pass them as arguments to packer, so temporary virtual machines can use private connections inside Vlan. 
+
+#### Service principal
+Packer will authorize in Azure infrastructure via Service Principal. If you want to prepare image-generation CI, or use packer manually, you need to create your SP with full read-write permissions for selected Azure subscription.
+Detailed instruction can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Prepare environment and image deployment
+#### How to prepare Windows build agent
+You can use local machine as build agent, or deploy virtual machine in [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/quick-create-cli) 
+
+Download `packer` from https://www.packer.io/downloads
+```
+$packerFolder = "C:\packer"
+Invoke-WebRequest -Uri https://releases.hashicorp.com/packer/1.6.0/packer_1.6.0_windows_amd64.zip -OutFile "$packerFolder/packer.zip" && Expand-Archive $packerFolder/packer.zip .
```

Comment:
May we could leave only choco way to install packer / azure cli.
Custom code lines can become invalid in future and packer 1.6.0 can become deprecated (maintenance concern). choco is reliable way to get the latest packer version. Otherwise, customers will download file packer site manually.

Reply:
Done.